### PR TITLE
design of core openid connect spec

### LINKED ISSUE #63: Planning and work breakdown for OpenID Connect

### PR DESCRIPTION
**Pull request by seanmonstar**
Closes #63

### CODE REVIEW

File: features/FxA-38-openid-connect/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Details
+
+[OpenID Connect][spec] is a small layer on top of OAuth2. We are already largely compatible with OAuth2, and so the only addition needed to make us compliant with OpenID Connect is to add a `openid` scope, and return an `id_token` when that scope is requested.
+
+The `id_token` is a [JWT][jwt], with the following example:
+
+```json
+{
+  "iss": "https://oauth.accounts.firefox.com",
```
Comment (rfk) [inline features/FxA-38-openid-connect/README.md]: Do we want to keep this on "oauth.accounts.firefox.com", or work on some merging of the auth and oauth as part of this feature?

**Comment by rfk** [comment]
I think we should also include some basic provider metadata discovery, per Sections 3 and 4 of [1].  A good measure of success for this project will be if a relier can integrate knowing only a single URL for our service, rather than the handful of inter-connected service URLs that they must currently be configured with.

[1] https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata

File: features/FxA-38-openid-connect/README.md (hunk 1 above)
Comment (seanmonstar) [inline features/FxA-38-openid-connect/README.md]: I think `accounts.firefox.com` would be the best. (I know the auth server is at the `api` subdomain, but I prefer without it.)

**Comment by seanmonstar** [comment]
@rfk added a section describing the configuration endpoint that we could add.

File: features/FxA-38-openid-connect/README.md (hunk 1 above)
Comment (rfk) [inline features/FxA-38-openid-connect/README.md]: FWIW it's always annoyed me that the `api` subdomain shows up in external facing things e.g. identity assertion issuer strings, so :+1:

**Comment by rfk** [comment]
A great overview, and I'm feeling better about the scope of this work seeing it all written out :-)

Let's merge this and we can iterate on any finer details as we progress.

**Pull request merged**